reactor3/src/test/java/com/linecorp/armeria/common/reactor3/RequestContextPropagationFluxTest.java

```diff
@@ -0,0 +1,684 @@
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.reactor3;
+
+import static com.linecorp.armeria.common.reactor3.ContextAwareMonoTest.ctxExists;
+import static com.linecorp.armeria.common.reactor3.ContextAwareMonoTest.newContext;
+import static com.linecorp.armeria.common.reactor3.ContextAwareMonoTest.noopSubscription;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import java.time.Duration;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.reactivestreams.Publisher;
+
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.common.RequestContextAccessor;
+import com.linecorp.armeria.common.util.SafeCloseable;
+import com.linecorp.armeria.internal.testing.AnticipatedException;
+import com.linecorp.armeria.internal.testing.GenerateNativeImageTrace;
+
+import reactor.core.Disposable;
+import reactor.core.publisher.ConnectableFlux;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Hooks;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+import reactor.test.StepVerifier;
+import reactor.util.context.Context;
+
+@GenerateNativeImageTrace
+class RequestContextPropagationFluxTest {
+
+    static Stream<Arguments> provideContextWriteAndCaptureTestCase() {
+        return Stream.of(
+                // shouldContextWrite, shouldContextCapture.
+                Arguments.of(true, false),
+                Arguments.of(false, true)
+        );
+    }
+
+    @BeforeAll
+    static void setUp() {
+        Hooks.enableAutomaticContextPropagation();
+    }
+
+    @AfterAll
+    static void tearDown() {
+        Hooks.disableAutomaticContextPropagation();
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideContextWriteAndCaptureTestCase")
+    void fluxError(boolean shouldContextWrite,
+                   boolean shouldContextCapture) {
+        final ClientRequestContext ctx = newContext();
+        final Flux<Object> flux;
+
+        flux = addCallbacks(Flux.error(() -> {
+            // This is called twice. after publishOn() and verifyErrorMatches()
+            // After publishOn(), ctxExists(ctx) should be false.
+            // On the other hand, it should be True due to ContextPropagation.
+            return new AnticipatedException();
```

Comment:
I think we still need to check the context if it works or not. How about:
```suggestion
        final AtomicBoolean atomicBoolean = new AtomicBoolean();
        flux = addCallbacks(Flux.error(() -> {
            if (!atomicBoolean.getAndSet(true)) {
                // Flux.error().publishOn() calls this error supplier immediately to see if it can retrieve
                // the value via Callable.call() without ctx.
                assertThat(ctxExists(ctx)).isFalse();
            } else {
                assertThat(ctxExists(ctx)).isTrue();
            }
            return new AnticipatedException();
```

Reply:
Thanks for your comment 👍 
I followed your guideline!